jdaviz/app.py

```diff
@@ -1162,18 +1162,22 @@ def simplify_spectral_subset(self, subset_name, att, overwrite=False):
             Whether to update the current subset with the simplified state or apply it
             to a new subset.
         """
-        spectral_region = self.get_subsets(subset_name, spectral_only=True)
-        new_state = None
-        # Reverse through sub regions so that they are added back
-        # in the order of lowest values to highest
-        for index in range(len(spectral_region) - 1, -1, -1):
-            convert_to_range = RangeSubsetState(spectral_region[index].lower.value,
-                                                spectral_region[index].upper.value,
-                                                att)
-            if new_state is None:
-                new_state = convert_to_range
-            else:
-                new_state = new_state | convert_to_range
+        if self.is_there_overlap_spectral_subset(subset_name):
+            new_state = self.merge_overlapping_spectral_regions(subset_name, att)
+        else:
+            new_state = None
+            spectral_region = self.get_subsets(subset_name, spectral_only=True)
+            # new_state = None
+            # Reverse through sub regions so that they are added back
+            # in the order of lowest values to highest
```

Comment:
I wonder if this will ever cause problems with inverted axes limits?  The only place we do that now is with the pixel axis in a 2d spectrum, so that _might_ be the cause of the problems I'm seeing in that case... but if it isn't, we can probably ignore for now.